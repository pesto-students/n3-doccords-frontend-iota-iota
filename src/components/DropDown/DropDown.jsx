import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
// import Button from "@material-ui/core/Button";

const useStyles = makeStyles((theme) => ({
  button: {
    display: "block",
    marginTop: theme.spacing(2),
  },
  formControl: {
    margin: theme.spacing(1),
    width: "100%",
  },
}));

const DropDown = () => {
  const classes = useStyles();
  const [healthTopic, setHealthTopic] = React.useState("");
  const [open, setOpen] = React.useState(false);

  const handleChange = (event) => {
    setHealthTopic(event.target.value);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleOpen = () => {
    setOpen(true);
  };

  return (
    <div>
      <FormControl className={classes.formControl}>
        <InputLabel id="demo-controlled-open-select-label">
          Select Health Topic
        </InputLabel>
        <Select
          labelId="demo-controlled-open-select-label"
          id="demo-controlled-open-select"
          open={open}
          onClose={handleClose}
          onOpen={handleOpen}
          value={healthTopic}
          onChange={handleChange}
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          <MenuItem value={10}>Cancer</MenuItem>
          <MenuItem value={20}>Blood Pressure</MenuItem>
          <MenuItem value={30}>Covid-19</MenuItem>
        </Select>
      </FormControl>
    </div>
  );
};

export default DropDown;
