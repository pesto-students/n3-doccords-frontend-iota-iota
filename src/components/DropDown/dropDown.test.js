import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import DropDown from "components/DropDown/DropDown";

describe("DropDown", () => {
  it("rendering without Error", () => {
    const wrapper = shallow(<DropDown />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
