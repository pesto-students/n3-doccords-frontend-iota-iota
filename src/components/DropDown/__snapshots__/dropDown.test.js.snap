// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown rendering without Error 1`] = `
<div>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-formControl-2"
  >
    <WithStyles(ForwardRef(InputLabel))
      id="demo-controlled-open-select-label"
    >
      Select Health Topic
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Select))
      id="demo-controlled-open-select"
      labelId="demo-controlled-open-select-label"
      onChange={[Function]}
      onClose={[Function]}
      onOpen={[Function]}
      open={false}
      value=""
    >
      <WithStyles(ForwardRef(MenuItem))
        value=""
      >
        <em>
          None
        </em>
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value={10}
      >
        Cancer
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value={20}
      >
        Blood Pressure
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value={30}
      >
        Covid-19
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </WithStyles(ForwardRef(FormControl))>
</div>
`;
